src/gpsw/utils.py

```
@@ -53,44 +54,150 @@ def dry_run(args):
+        f.write("    margin=0.2;\n")
+        f.write(
+            '    node [shape=box, style="rounded,filled", fontname=Helvetica, fontsize=10, penwidth=1.5, color=black, fillcolor=white];\n'
+        )
+        f.write('    edge [color="#888888", penwidth=1.2];\n\n')
+
+        for category, nodes in clusters.items():
+            fill = category_colors.get(category, "#F5F5F5")
+            safe_id = re.sub(r"\W+", "_", category.lower())
+            f.write(f"    subgraph cluster_{safe_id} {{\n")
+            f.write(f'        label = "{category}";\n')
+            f.write(f'        style = "filled,dashed";\n')
```

Ruff (0.11.9)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)